pkg/utils/retry/retry.go

```diff
@@ -0,0 +1,61 @@
+package retry
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/jpillora/backoff"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+)
+
+type ctxKey string
+
+// CtxKeyRetryID is the context key for tracing ID
+const CtxKeyRetryID ctxKey = "retryID"
```

Comment:
Can probably drop `Retry` from here too :thinking: 
```suggestion
// CtxKey is the context key for retry ID
const CtxKey ctxKey = "retryID"
```
Actually, a helper for setting the value might be more useful, in order to make sure the correct type (string) is used. The current code would overwrite the value if it was the wrong type.
```suggestion
// CtxWithID sets the retry ID context key for logging.
func CtxWithID(ctx context.Context, retryID string) context.Context {
	return context.WithValue(ctx, ctxKey, retryID)
}
```

Reply:
refactored this.